File: presentation/src/main/java/io/vladkarpman/presentation/details/view/MovieViewHolder.java
package io.vladkarpman.presentation.details.view;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.UiThread;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import io.vladkarpman.presentation.R;
import io.vladkarpman.domain.entity.Movie;

public class MovieViewHolder extends RecyclerView.ViewHolder {

    private final TextView tvTitle;
    private final TextView tvOverview;
    private final ImageView ivPoster;

    public MovieViewHolder(@NonNull View itemView) {
        super(itemView);
        tvTitle = itemView.findViewById(R.id.tvTitle);
        tvOverview = itemView.findViewById(R.id.tvOverview);
        ivPoster = itemView.findViewById(R.id.ivPoster);
    }

    @UiThread
    public void bind(@NonNull Movie movie) {
        tvTitle.setText(movie.getTitle());
        tvOverview.setText(movie.getDescription());
        // TODO: 10/30/20 do it more generically
        Glide.with(itemView.getContext()).load(movie.getPosterUrl()).into(ivPoster);
    }
}

File: app/src/main/java/io/shelfy/architecture/ui/adapters/MovieViewHolder.kt
package io.shelfy.architecture.ui.adapters

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import io.shelfy.architecture.domain.model.Movie
import coil.api.load
import kotlinx.android.synthetic.main.item_movie.view.*

class MovieViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    fun bind(movie: Movie, onClick: () -> Unit) {
        with(itemView) {
            with(movie) {
                tvTitle.text = title
                tvOverview.text = description
                ivPoster.load(posterUrl)
            }
            setOnClickListener { onClick() }
        }
    }
}
